CODEC-285-JUnit5

itamarc opened the pull request:
I migrated all tests in commons-coded to use JUnit5.
I also dropped junit-vintage-engine from pom.xml, since it's not needed anymore.

kinow reviewed (commented):
Hi @itamarc (olá! :wave: )

You'll have to rebase your branch, looks like there are some conflicts now :+1: After that we will likely squash the commits down to a single commit or maybe a couple meaningful commits (it'd be better if you could do it).

Thanks!
Bruno

itamarc commented:
Hi, @kinow (olá)

I can work on solving the conflicts and also squash the commits to one, but...

1) I don't know how to see the conflicts, I can only see the list of files with conflicts. Maybe it's because I don't have write permission on commons-codec repository:

> **This branch has conflicts that must be resolved**
> Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.
> **Conflicting files**
> (...)

How can I see the conflicts so I can solve them?

2) After solving the conflicts, can I squash the commits on my own branch and it will show up on the PR?

Sorry, it's my first PR of this size. And thank you in advance for your help.

Itamar

aherbert commented:
You can fetch the master and then rebase on that. On the command line:
```
git fetch origin/master
git rebase origin/master
```
The rebase attempts to collect all your commits and apply them sequentially to the master. If this is not possible the rebase will warn about the conflicts. You then fix those locally and continue with the rebase, IIRC `git rebase --continue`.

Eventually you should end up with a set of changes that can be applied directly onto the master branch. You then force push the changes to this PR branch to update it: `git push -f`.

garydgregory commented on src/test/java/org/apache/commons/codec/BinaryEncoderAbstractTest.java:
```diff
@@ -34,13 +35,8 @@ public void testEncodeEmpty() throws Exception {
-        } catch (final EncoderException ee) {
-            // An exception should be thrown
-        }
+        assertThrows(EncoderException.class, () -> { encoder.encode(null); });
```
@itamarc 
Thank you for your PR.
You do not need the curly brackets here, less clutter is better.

garydgregory commented on src/test/java/org/apache/commons/codec/StringEncoderAbstractTest.java:
```diff
@@ -72,16 +67,10 @@ public void testEncodeNull() throws EncoderException {
-        }
-        Assert.assertTrue("An exception was not thrown when we tried to encode " + "a Float object", exceptionThrown);
+        final StringEncoder encoder = this.getStringEncoder();
+        assertThrows(EncoderException.class, () -> { encoder.encode(Float.valueOf(3.4f)); },
```
No curlies needed; I'll assume you'll go over the other files. I'll review again after you push.

itamarc replied:
I tried to not change or remove any of the messages (except for typos), even if I think the message was not necessary. All I did was to change the position of the message as a parameter, since this changed from JUnit 4 to 5. My intention was to minimize the changes, and afterall, some of the messages give information about the failure of the test that the message from JUnit will not give.

garydgregory replied:
My comment was not about assertion messages, but about the extra {} that are superfluous.

itamarc replied:
Yes, sorry not letting this clear. I understood and will remove them. I only commented about the messages because when I started the rebase process (as instructed by @aherbert), I noticed that some messages were removed.

garydgregory commented on src/test/java/org/apache/commons/codec/binary/Base16InputStreamTest.java:
```diff
@@ -327,33 +323,21 @@ public void testReadOutOfBounds() throws IOException {
-            } catch (final IndexOutOfBoundsException e) {
-                // Expected
-            }
+            assertThrows(IndexOutOfBoundsException.class, () -> {
```
@itamarc 
Same comment as for the other extra {} in other files...
TY!

itamarc commented:
I made the rebase with the origin/master and looks like I solved all conflicts and I just squashed the commits into 1.

I also removed the extra curly braces like pointed by @garydgregory

daniel-tavares-osf reviewed (commented):
Very nice work. Congratulations.
